Docs/ArchitectureDecisions/004_observability.md

```diff
@@ -0,0 +1,102 @@
+# 003 - Observability
+
+**Date**: 2021-11-08
+
+**Participants**:  Patrick Schuler, Bastian Burger, Eugene Fedorenko
+
+## Status
+
+Accepted
+
+## Context
+
+The goal of observability for the LoRaWAN IoT Edge starter kit is to:
+
+1. Monitor if the LoRaWAN Starter Kit solution works according to the user expectations regarding the following factors:
+   - Coverage. The data is coming from the majority of observed IoT assets
+   - Freshness. The data coming from the assets is fresh and relevant
+   - Throughput. The data is delivered from the assets without significant delays.
+   - Correctness. The ratio of errors and lost messages from the assets is small
+1. Provide monitoring instruments to detect possible failure/violation in each factor
+1. Provide instruments to identify and diagnose failures to get to the problem quickly
+
+The decisions in the following will apply to our LoRaWAN Network Server (LNS) implementation.
+
+## Decisions
+
+We will support Azure Monitor as a first-class monitoring solution for our starter kit. A user can opt-in to use Application Insights with the starter kit, in which case we will support a rich set of observability features. If the user decides to not use Application Insights, we will still support essential monitoring capabilities. This means that we will:
+
+- Track LNS logs in Application Insights (when opted in). We will adhere to the [IoT Edge recommended format](https://docs.microsoft.com/en-gb/azure/iot-edge/how-to-retrieve-iot-edge-logs?view=iotedge-2020-11#recommended-logging-format) for the structure of the log console output. Export of logs to anything else than Application Insights requires a custom solution by the user and is not supported by the starter kit.
+- Always expose metrics using [prometheus-net](https://github.com/prometheus-net/prometheus-net).
+  - Additionally, we track LNS metrics using the ASP.NET Core Application Insights SDK (when opted in)
+- Track traces using the Application Insights SDK (when opted in)
+- Support alerts when using Application Insights
```

Comment:
Support alerts when using Application Insights and/or Log Analytics (with Prometheus format and metrics collector module)

Reply:
Done with 0473b66eea2406aa0a9b1cc346e8891dc587f6c0